FalloutTerminalKeyMapper/AppDelegate.m

```diff
@@ -18,63 +18,189 @@ @interface AppDelegate ()
-    NSArray *soundFiles = @[@"k2", @"k3", @"k4"];
-    
-    [NSEvent addGlobalMonitorForEventsMatchingMask:(NSKeyDownMask) handler:^(NSEvent *event){
-        if (!self.isMuted) {
-            uint32_t random_index = arc4random_uniform((uint32_t)soundFiles.count);
-            NSString *resoucePath = [[NSBundle mainBundle] pathForResource:event.keyCode != 36 ? soundFiles[random_index] : @"kenter" ofType:@"mp3"];
-            NSSound *sound = [[NSSound alloc] initWithContentsOfFile:resoucePath byReference:YES];
-            [sound play];
-        }
-        
-        //Toggle Mute
-        if (([event modifierFlags] & NSCommandKeyMask) && ([event modifierFlags] & NSShiftKeyMask) && (event.modifierFlags & NSControlKeyMask) && event.keyCode == 3) {
-            self.isMuted = !self.isMuted;
-        }
-    }];
-    
-    [[[NSWorkspace sharedWorkspace] notificationCenter] addObserver:self
-                                                           selector:@selector(terminalLaunched:)
-                                                               name:NSWorkspaceDidLaunchApplicationNotification
-                                                             object:nil];
-    
-    [[[NSWorkspace sharedWorkspace] notificationCenter] addObserver:self
-                                                           selector:@selector(terminalTerminated:)
-                                                               name:NSWorkspaceDidTerminateApplicationNotification
-                                                             object:nil];
-    
-    self.statusItem = [[NSStatusBar systemStatusBar] statusItemWithLength:NSVariableStatusItemLength];
-    [self.statusItem setMenu:self.statusMenu];
-    [self.statusItem setImage:[NSImage imageNamed:@"pipboy_icon"]];
-    [self.statusItem setHighlightMode:YES];
+	self.isMuted = NO;
+	self.supportedBundles = @[@"com.googlecode.iterm2", @"com.apple.Terminal", @"io.cool-retro-term"];
+	NSArray *soundFiles = @[@"k2", @"k3", @"k4"];
+	
+	[NSEvent addGlobalMonitorForEventsMatchingMask:(NSKeyDownMask) handler:^(NSEvent *event){
+		
+		BOOL terminalIsActive = NO;
+		
+		// Keyboard shortcuts
+		if (([event modifierFlags] & NSCommandKeyMask) && ([event modifierFlags] & NSShiftKeyMask) && (event.modifierFlags & NSControlKeyMask)) {
+			if (event.keyCode == kVK_ANSI_K) {		//Toggle Mute
+				[self toggleMute];
+			}else if (event.keyCode == kVK_ANSI_L){		//Toggle Terminals Only setting
+				[self togglePlayMode];
+			}
+		}
+		
+		BOOL onlyPlayOnTerminal = [[[NSUserDefaults standardUserDefaults] objectForKey:@"terminalsOnly"] boolValue];
+		if (onlyPlayOnTerminal) { //Determine if terminal is active
+			for (NSRunningApplication *currApp in [[NSWorkspace sharedWorkspace] runningApplications]) {
+				if ([currApp isActive] && [self.supportedBundles containsObject:currApp.bundleIdentifier]) {
+					
+					terminalIsActive = YES; break;
```

Comment:
Thanks for your opinion. I'm always interested in checking out other coders styles. In any case, I definitely feel your code fit well. Thanks again for contributing. If you have any other ideas for how to make this app better and would like to collaborate with me, feel free to make more PR's or email me directly at hagonzalez94@gmail.com